buildSrc/src/main/kotlin/buildsrc/tasks/GenerateYamlTestSuiteData.kt

```diff
@@ -0,0 +1,259 @@
+            // multiple cases, the case number.
+
+            val dirHasMultipleCases = parentDirContents.all {
+                it.isDirectory && it.name.toIntOrNull() != null
+            }
+
+            val id =
+                if (dirHasMultipleCases) {
+                    "${dir.parentFile.name}:${dir.name}"
+                } else {
+                    dir.name
+                }
+            YamlTestSuiteDirSpec(id = id, dir = dir)
+        }
+
+        yamlTestSuiteDirsById.forEach { (id, testCaseDir, objectName) ->
+            // error -- This file indicates the YAML should fail to parse
+            val expectError = testCaseDir.resolve("error").exists()
+
+            val contents = if (expectError) {
+                generateExpectErrorData(
+                    objectName = objectName,
+                    id = id,
+                    testCaseDir = testCaseDir,
+                )
+            } else {
+                generateExpectSuccessData(
+                    objectName = objectName,
+                    id = id,
+                    testCaseDir = testCaseDir,
+                )
+            }
+
+            val destFile = destination.resolve("$objectName.kt")
+
+            destFile.writeText(contents)
+        }
+
+        val allTestSuiteIdListContents = yamlTestSuiteDirsById
+            .map { (_, _, objectName) -> "$objectName.id to $objectName," }
+            .sorted()
+            .joinToString("\n")
+            .prependIndent("  ")
+
+        val yamlTestSuiteValuesFile = temporaryDir.resolve("yamlTestSuiteValues.kt")
+
+        // use `internal object` instead of `internal val` to try and avoid https://youtrack.jetbrains.com/issue/KT-59274
+        yamlTestSuiteValuesFile.writeText(/* language=kotlin */ """
+            ¦package org.snakeyaml.engine.test_suite
+            ¦
+            ¦/** All YAML test suite data */
+            ¦internal object YamlTestSuiteData : Map<YamlTestData.Id, YamlTestData> by mapOf(
+            ¦$allTestSuiteIdListContents
+            ¦)
+            ¦
+        """.trimMargin(/* language=text */ "¦")
+        )
+    }
+
+    private fun generateExpectErrorData(
+        objectName: String,
+        id: String,
+        testCaseDir: File,
+    ): String {
+        // parse the data in the directory...
+
+        // === -- The name/label of the test
+        val label = testCaseDir.findSuiteName()
+            ?: error("missing === label in $testCaseDir")
+        // in.yaml -- The YAML input to be parsed or loaded
+        val inYaml = testCaseDir.resolve("in.yaml").readTextOrNull()
+            ?: error("missing in.yaml in $testCaseDir")
+
+        val relativePath = testCaseDir.relativeTo(rootDir).invariantSeparatorsPath
+
+        return /* language=kotlin */ """
+            ¦|¦package org.snakeyaml.engine.test_suite
+            ¦|¦
+            ¦|¦/**
+            ¦|¦ * Test case for file `./$relativePath`
```

Comment:
I'm a bit surprised we're generating code here - was it strictly necessary? I can imagine we just download the YAMLs (test cases), and build the unit test cases on the fly. Just asking out of curiosity, I'm not strictly against it, just want to understand this decision better + quickly discuss pros/cons 😄

Reply:
Code generation isn't strictly necessary, but it's the easiest way to access files. There's no easy Kotlin Multiplatform cross-platform file system (Okio doesn't work on JS, and something like kotlinx-resources is [missing some features](https://github.com/goncalossilva/kotlinx-resources/issues/65)).

Reply:
Ah, right! I subconsciously assumed we can read files for all platforms, because Node/JS can do that. Cool, it's a good enough explanation to me. How about either adding a short comment here, or adding a top-level DECISION_LOG.md and describing it shortly there?

Reply:
I added a comment here https://github.com/krzema12/snakeyaml-engine-kmp/pull/59/files#diff-4267df28c53bd234c9e38408244193cedab5edfb3d2bfb1b9d22a4bab2502cc5R17-R18

I'm not a fan of introducing docs that aren't 'local' to the actual code, or aren't automatically tested (should out to [Knit](https://github.com/Kotlin/kotlinx-knit/)), but if you think it would help then go for it.

Reply:
Ok, it's good enough, thanks 😃